layouts/shortcodes/ecosystem/registry/search-form.html

```diff
@@ -100,11 +111,174 @@
+      let endPage = Math.min(totalPages, startPage + pagesToShow - 1);
+
+      // Adjust if there are not enough pages before the current page
+      if (startPage > 1) {
+        const diff = currentPage - startPage;
+        if (endPage + diff < totalPages) {
+          endPage += diff; // Expand the end if space allows
+        } else {
+          startPage = Math.max(1, totalPages - pagesToShow + 1); // Adjust start if needed
+        }
+      }
+
+      // Create page number buttons
+      for (let i = startPage; i <= endPage; i++) {
+        const pageButton = document.createElement('button');
+        pageButton.classList.add('btn', 'btn-outline-secondary', 'mx-1');
+        pageButton.textContent = i;
+        pageButton.addEventListener('click', () => {
+          currentPage = i;
+          showPage(currentPage);
+        });
+
+        // Highlight the current page button
+        if (i === currentPage) {
+          pageButton.classList.add('active');
+        }
+
+        paginationNumbers.appendChild(pageButton);
+      }
+    }
+
+    // Update pagination buttons
+    updateVisiblePages(page);
+
+    // Disable/enable prev/next buttons
+    document.getElementById('prev-page').disabled = page === 1;
+    document.getElementById('next-page').disabled = page === totalPages;
+
+    // Update total entries displayed
+    document.getElementById('pagination-numbers').textContent = `Page ${page} of ${totalPages}`;
+  }
+
+  // Initialize and show first page
+  updateTotalPages(); // Call this to update the total pages count
+  showPage(currentPage);
+
+  // Add event listeners for pagination buttons
+  document.getElementById('prev-page').addEventListener('click', () => {
+    if (currentPage > 1) {
+      currentPage--;
+      showPage(currentPage);
+    }
+  });
+
+  document.getElementById('next-page').addEventListener('click', () => {
+    if (currentPage < totalPages) {
+      currentPage++;
+      showPage(currentPage);
+    }
+  });
+
+  // Function to handle "Go to Page" input
+  function goToPage() {
+    const pageInput = document.getElementById('page-input').value;
+    const page = parseInt(pageInput);
+    if (!isNaN(page) && page >= 1 && page <= totalPages) {
+      currentPage = page;
+      showPage(currentPage);
+    } else {
+      document.getElementById('page-error').textContent = 'Invalid page number';
+    }
+  }
+
+
+  function updateCurrentPageDisplay(currentPage, totalPages) {
+    const pageDisplay = document.getElementById('page-display');
+    pageDisplay.textContent = `Page ${currentPage} of ${totalPages}`;
+  }
+  // Function to handle results per page selection
+  function changeResultsPerPage() {
+    const selectedValue = document.getElementById('results-per-page').value;
+    resultsPerPage = parseInt(selectedValue);
+    currentPage = 1; // Reset to first page when changing results per page
+    updateTotalPages(); // Update total pages and show first page
+  }
+</script>
```

Comment:
using inline javascript is not recommended, move this into a dedicated file, I am OK with this being a part of https://github.com/open-telemetry/opentelemetry.io/blob/main/assets/js/registrySearch.js